dispatch/static/manager/src/js/components/fields/WidgetField.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import R from 'ramda'
+
+import Panel from '../Panel'
+import WidgetSelectInput from '../inputs/WidgetSelectInput'
+import WidgetFieldWrapper from '../ZoneEditor/WidgetField'
+
+export default class WidgetFieldComponent extends React.Component {
+  handleWidgetChange(widgetId) {
+    this.props.onChange({
+      id: widgetId,
+      data: {}
+    })
+  }
+
+  updateField(name, data) {
+    const widgetData = {}
+    widgetData[name] = data
+
+    this.props.onChange(R.merge(this.props.data || {}, {
+      data: R.merge(this.props.data.data || {}, widgetData)
+    }))
+  }
+
+  getWidgetId() {
+    return R.prop('id', this.props.data || {})
```

Comment:
You can make use of [default props](https://facebook.github.io/react/docs/typechecking-with-proptypes.html#default-prop-values) to get rid of all the `this.props.data || {}` expressions throughout this class - makes things a bit more readable.